lsstseries/analysis/stetsonj.py

```diff
@@ -0,0 +1,139 @@
+    Returns
+    -------
+    stetsonJ : `float`
+        StetsonJ statistic
+    References
+    ----------
+    .. [1] Stetson, P. B., "On the Automatic Determination of Light-Curve
+    Parameters for Cepheid Variables", PASP, 108, 851S, 1996
+    Notes
+    ----------
+    Taken from
+    https://github.com/lsst/meas_base/blob/main/python/lsst/meas/base/diaCalculationPlugins.py
+    Using the function on random gaussian distribution gives result of -0.2
+    instead of expected result of 0?
+    """
+    mean = None
+    n_points = len(fluxes)
+    if n_points <= 1:
+        return np.nan
+    flux_mean = _stetson_mean(fluxes, errors, mean)
+    delta_val = (
+        np.sqrt(n_points / (n_points - 1)) * (fluxes - flux_mean) / errors)
+    p_k = delta_val ** 2 - 1
+    return np.mean(np.sign(p_k) * np.sqrt(np.fabs(p_k)))
+
+def _stetson_mean(values,
+                  errors,
+                  mean=None,
+                  alpha=2.,
+                  beta=2.,
+                  n_iter=20,
+                  tol=1e-6):
+    """Compute the stetson mean of the fluxes which down-weights outliers.
+    Weighted biased on an error weighted difference scaled by a constant
+    (1/``a``) and raised to the power beta. Higher betas more harshly
+    penalize outliers and ``a`` sets the number of sigma where a weighted
+    difference of 1 occurs.
+    Parameters
+    ----------
+    values : `numpy.dnarray`, (N,)
+        Input values to compute the mean of.
+    errors : `numpy.ndarray`, (N,)
+        Errors on the input values.
+    mean : `float`
+        Starting mean value or None.
+    alpha : `float`
+        Scalar down-weighting of the fractional difference. lower->more
+        clipping. (Default value is 2.)
+    beta : `float`
+        Power law slope of the used to down-weight outliers. higher->more
+        clipping. (Default value is 2.)
+    n_iter : `int`
+        Number of iterations of clipping.
+    tol : `float`
+        Fractional and absolute tolerance goal on the change in the mean
+        before exiting early. (Default value is 1e-6)
+    Returns
+    -------
+    mean : `float`
+        Weighted stetson mean result.
+    References
+    ----------
+    .. [1] Stetson, P. B., "On the Automatic Determination of Light-Curve
+    Parameters for Cepheid Variables", PASP, 108, 851S, 1996
+    Notes
+    ----------
+    Taken from
+    https://github.com/lsst/meas_base/blob/main/python/lsst/meas/base/diaCalculationPlugins.py
+    """
+    n_points = len(values)
+    n_factor = np.sqrt(n_points / (n_points - 1))
+    inv_var = 1 / errors ** 2
+    if mean is None:
+        mean = np.average(values, weights=inv_var)
+    for iter_idx in range(n_iter):
+        chi = np.fabs(n_factor * (values - mean) / errors)
+        tmp_mean = np.average(
+            values,
+            weights=inv_var / (1 + (chi / alpha) ** beta))
+        diff = np.fabs(tmp_mean - mean)
+        mean = tmp_mean
+        if diff / mean < tol and diff < tol:
```

Comment:
There is a potential divide by zero here if mean is 0.0.

Reply:
I am torn about this one. On the one hand I would not want to touch ported lsst code. This is only a problem if values are 0, in which case mean is 0 as well. That should never be the case with the lsst data. 
Having said that, I am also uncomfortable about having this obvious point of possible failure here. One way to solve this is to add condition to if statement to break also if mean == 0?

Reply:
+1 to Neven's suggestion, since the potential divide by zero is in a condition that just determines whether or not to break, adding another break condition seems appropriate.